kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/ImageServiceBase.java

```diff
@@ -137,18 +137,22 @@ public String getActiveProcessImage(String containerId, long procInstId, String
 
             for (NodeInstanceDesc activeNode : activeLogs) {
                 active.put(activeNode.getId(), activeNode.getNodeId());
-
-                populateSubProcessLink(containerId, activeNode, subProcessLinks);
             }
 
             for (NodeInstanceDesc completeNode : completedLogs) {
                 completed.add(completeNode.getNodeId());
 
                 active.remove(completeNode.getId());
-
-                populateSubProcessLink(containerId, completeNode, subProcessLinks);
             }
 
+            completedLogs.forEach(completedNode -> {
+                populateSubProcessLink(containerId, completedNode, subProcessLinks);
```

Comment:
@bxf12315 I guess this can be done in the previous loop, not necessary a new one.

Reply:
you can see @jstastny-cz 's comment, I think a better way is to keep old order.

Reply:
@bxf12315 I agree to keep  the order of the loops, I guess what you are trying is to populateSubProcessLink of active nodes after the completed ones. My comment was regarding the need of the new loop to populateSubProcess of complete nodes when you can keep that as it was done before in the previous loop.

Reply:
Done